nixos/modules/image/repart.nix

```diff
@@ -90,8 +90,10 @@ in
     };
 
     package = lib.mkPackageOption pkgs "systemd-repart" {
-      default = "systemd";
-      example = "pkgs.systemdMinimal.override { withCryptsetup = true; }";
+      # We use buildPackages so that repart images are built with the build
+      # platform's systemd, allowing for cross-compiled systems to work.
+      default = [ "buildPackages" "systemd" ];
```

Comment:
Now that I read this again,  do we need this to be buildPackages? Shouldn't callPackage splice this correctly itself?

Reply:
I don't think so, since we have the package option it gets passed to image-repart.nix directly, removing any splicing that would have been done for the `systemd` argument

Reply:
Ok, I'll take your word for it.